### ISSUE TITLE
Implement Snapd

### ISSUE THREAD

**Issue by elementaryBot** [elementary/appcenter#27]
Informational.

If we're going to implement Snappy support we'll want to use: https://launchpad.net/snapd-glib

According to Robert:
* Implements all of the snapd REST API: https://github.com/snapcore/snapd/blob/master/docs/rest.md
* Provides both synchronous and asynchronous methods
* Supports Vala
* Will SRU to Ubuntu 14.04 once the API/ABI is stable.
* Uploaded to Ubuntu Yakkety, will be there once it is accepted into the NEW queue. Use the libsnapd-glib-dev or gir1.2-snapd-0 packages.

Launchpad Details: [#LP1614844](https://bugs.launchpad.net/bugs/1614844) Daniel Fore - 2016-08-19 06:38:53 +0000

**Comment by peteruithoven** [comment]
( This is being implementing into a separate app: https://github.com/bartzaalberg/snaptastic )

**Comment by jlnr** [comment]
@peteruithoven Does that mean that Snap support is not coming to App Center? Being able to install Spotify, IntelliJ etc. right from the primary app store would be fantastic.

**Comment by peteruithoven** [comment]
I don't know what will happen, just wanted to point to snaptastic.

**Comment by danrabbit** [comment]
I'm going to close this since it seems nobody has an interest in doing this and our plan for AppCenter (the service) is to use Flatpak.

I think, however, if someone decided to implement snap support we probably wouldn't be opposed to that :man_shrugging:

**Comment by freechelmi** [comment]
Just a note : 

- Some of our customers want eOS preinstalled
- Some of the software they want preinstalled are only available on the snapstore, not as flatpak from flathub.


So Snap support is really a plus for those newcomers.

**Comment by cassidyjames** [comment]
@freechelmi I understand that, and if Snap support were contributed to AppCenter, I _think_ we'd be open to it as @danrabbit has said. But we don't have the resources to commit to implementing it ourselves, especially when it doesn't align with our requirements—and there is a concern with shipping a third-party controlled remote repository of software. Note that we have chosen not to ship Flathub out of the box for similar concerns. So it's not a guarantee without quite a bit more thought.

For context, we talked a bit about why we chose Flatpak, including choosing it over other options here: https://blog.elementary.io/elementary-appcenter-flatpak

Another route that might be more fruitful than implementing (and then perpetually maintaining) snap support in AppCenter could be talking to those app developers to see if we can get their software into Flathub. If that's not successful, then Flathub folks are pretty liberal with packaging up third-party software that is okay to redistribute and hosting it on Flathub. As of elementary OS 5.1, Flathub installs are _very_ streamlined with our Sideload utility.
